forem/forem
# Fix small bugs related to identities when merging a user

## What type of PR is this? (check all applicable)

## Description
This fixes a bug with merging users where having non-unique-by-provider identities prevent two accounts from being merged. This PR will allow this scenario to merge successfully:
1. the user signed up with Twitter
2. the user made another account with GitHub

It also refactors some old code a bit and gives a friendly warning if both accounts have same provider identities (Twitter and Twitter for example).

## Related Tickets & Documents
https://github.com/forem/internalCommunity/issues/54
https://github.com/forem/internalCommunity/issues/188

## QA Instructions, Screenshots, Recordings
1. Sign up for an account with Twitter
2. Logout
3. Sign up for an account via GitHub
4. Logout
5. Log into your admin account
6. Merge the two accounts via http://localhost:3000/admin/users/:id/edit
7. The merge should be successful!

### UI accessibility concerns?
Nope

## Added tests?


## [Forem core team only] How will this change be communicated?


## [optional] Are there any post deployment tasks we need to perform?
No

## [optional] What gif best describes this PR or how it makes you feel?

![A butterfly lands on a bunny's nose!](https://media.giphy.com/media/wSl9AXqgXLMLK6ijEa/source)
